server/src/test/java/org/elasticsearch/search/query/QueryPhaseTests.java

```diff
@@ -357,9 +357,9 @@ public void testInOrderScrollOptimization() throws Exception {
     public void testTerminateAfterSize0HitCountShortcut() throws Exception {
         indexDocs();
         // TotalHitCountCollector (used when size==0) shortcuts total hit count (using Weight#count) segment by segment.
-        // EarlyTerminatingCollector checks whether the threshold is reached at the beginning of each leaf collection.
-        // That's why the returned total hit count may be higher than terminate_after (or track_total_hits), yet lower
-        // than the number of docs depending on number of segments and documents distribution.
+        // With shortcut enabled, PartialHitCountCollector will figure out that the threshold is reached at the beginning
+        // of the following leaf collection. That's why the returned total hit count may be higher than terminate_after
+        // (or track_total_hits), yet lower than the number of docs depending on number of segments and documents distribution.
```

Comment:
I wonder if this comment would make more sense as part of the collectors rather than this test. This answers the first question that came to mind when I lookeda the collectors.

Reply:
Thanks, I added this to the javadocs, I also realized I needed to add a unit test at least for PartialHitCountCollector, which revealed a bug, and the corresponding fix required updating some existing tests :)